### ISSUE TITLE
Test-Json cannot parse schema

### ISSUE THREAD

**Issue by Liturgist** [PowerShell/PowerShell#12873]
It appears that both the JSON data file and schema file are valid.
Is there any way to get more information about why the schema cannot be parsed?

# Steps to reproduce
```
PS C:\src\t\json> Test-Json -Json (Get-Content -Path .\project-manifest.json -Raw)
True
PS C:\src\t\json> Test-Json -Json (Get-Content -Path .\project-manifest.schema.json -Raw)
True
PS C:\src\t\json> Test-Json -Json (Get-Content -Path .\project-manifest.json -Raw) -Schema .\project-manifest.schema.json
Test-Json: Cannot parse the JSON schema.
PS C:\src\t\json> Test-Json -Json (Get-Content -Path .\project-manifest.json -Raw) -Schema .\project-manifest.schema.json -Verbose
Test-Json: Cannot parse the JSON schema.
```
# Expected behavior
```
PS C:\src\t\json> Test-Json -Json (Get-Content -Path .\project-manifest.json -Raw) -Schema .\project-manifest.schema.json
True
```
# Actual behavior
```
PS C:\src\t\json> Test-Json -Json (Get-Content -Path .\project-manifest.json -Raw) -Schema .\project-manifest.schema.json
Test-Json: Cannot parse the JSON schema.
```
# Environment data
```
PS C:\src\t\json> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.0.1
PSEdition                      Core
GitCommitId                    7.0.1
OS                             Microsoft Windows 10.0.18363
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

**Comment by adityapatwardhan** [comment]
@Liturgist The value for the `-Schema` is a string containing the schema. Please have a look at the example in the documentation. https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.utility/test-json?view=powershell-7#examples

Also, there is a recent PR which will bring in support for `-SchemaFile`.

**Comment by Liturgist** [comment]
@adityapatwardhan - Thank you for your quick reply. I should have thought of that. Yes, I should have read the examples as well.

Perhaps a good alias for the `-Schema` parameter would be `-SchemaString`. That might go with an alias for `-Json` as `-JsonString`.

Does the proposed change include `-JsonFile` and `-SchemaFile`?

**Comment by iSazonov** [comment]
`SchemaFile` was already added yesterday.

**Comment by Liturgist** [comment]
Does that mean that `-JsonFile` will not be considered?

**Comment by iSazonov** [comment]
You could open new issue with the proposal.

**Comment by HSweta** [comment]
Hey, I am trying to use Test-Json but apparently, it's showing the error cannot parse schema json
Please let me know what's the issue

```
address.json
{
  "definitions": {
     "required": [
      "street_address",
      "city",
      "state"
    ],
      "street_address": {
        "type": "string"
      },
      "city": {
        "type": "string"
      },
      "state": {
        "type": "string"
      }
    } 
  
}
```
subschema.json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
  "address":{
      "$ref":"address.json#/definitions"
    
  }
}
}

Powershell Script


'{"address":{"city": "ABCDEFGHIJKLMNOPQRSTUVWXY","state": "ABCDEFGHIJKLMNOPQRSTUVWXYZABC"}}' | Test-Json -SchemaFile 'demoschema.json'

**Comment by iSazonov** [comment]
@HSweta I hope this will be fixed in #18141. Please vote there.

**Comment by HSweta** [comment]
@iSazonov even getting the same error with draft-7

**Comment by iSazonov** [comment]
@HSweta It seems you use a refence to external file in the schema file - it is blocked by _default_ in modern standard. I guess we will have a parameter/switch in Test-Json to explicitly enable this.
